gamestonk_terminal/portfolio/portfolio_model.py

```diff
@@ -330,22 +341,91 @@ def get_rolling_beta(
     res = df.div(df.sum(axis=1), axis=0)
     res = res.fillna(0)
     comb = pd.merge(
-        hist["Close"], mark["Market"], how="left", left_index=True, right_index=True
+        hist["Close"], mark["Market"], how="outer", left_index=True, right_index=True
     )
     comb = comb.fillna(method="ffill")
-    df_var = comb.rolling(252).var().unstack()["Close"].to_frame(name="var")
     for col in hist["Close"].columns:
-        df1 = (
-            comb.rolling(252).cov().unstack()[col]["Close"].to_frame(name=f"cov_{col}")
-        )
-        df_var = pd.merge(df_var, df1, how="left", left_index=True, right_index=True)
-        df_var[f"beta_{col}"] = df_var[f"cov_{col}"] / df_var["var"]
-    final = pd.merge(res, df_var, how="left", left_index=True, right_index=True)
-    final = final.fillna(method="ffill")
-    final = final.drop(columns=["var"] + [f"cov_{x}" for x in uniques])
+        exog = sm.add_constant(comb["Close"])
+        rols = RollingOLS(comb[col], exog, window=252)
+        rres = rols.fit()
+        res[f"beta_{col}"] = rres.params["Close"]
```

Comment:
Just curious the reason for doing this instead of what you had? The formula for a regression coefficient is the same cov(x,y)/var(x) you calculated before.

Reply:
Yup its the exact same equation! Just thought that the package was a little more official and probably has built in error handling that I don't want to worry about doing myself. Also allows for more analysis in the future.